pos_invoicing/models/pos_order.py

```diff
@@ -15,3 +17,57 @@ def _prepare_invoice(self):
             'pos_pending_payment': True,
         })
         return res
+
+    @api.model
+    def create_from_ui(self, orders):
+        return super(
+            PosOrder, self.with_context(pos_order_invoiced=True)
+        ).create_from_ui(orders)
+
+    @api.multi
+    def action_pos_order_invoice(self):
+        # this function is called by the create_from_ui
+        # function, and in that case, invoice is then confirmed
+        # but if called directly (by UI, in the back-office)
+        # the confirmation is not done.
+        # So we confirm the invoice in that case.
+        res = super().action_pos_order_invoice()
+        if not self.env.context.get("pos_order_invoiced", False):
+            for order in self:
+                order.invoice_id.sudo().with_context(
+                    force_company=self.env.user.company_id.id,
+                    pos_picking_id=order.picking_id,
+                ).action_invoice_open()
+                # Check if total amount are the same
+                prec = order.invoice_id.currency_id.rounding
+                if float_compare(
+                    order.invoice_id.amount_total_signed - order.amount_total,
+                    0.0,
+                    precision_digits=prec
+                ):
```

Comment:
That works, but this is how float_compare is supposed to be used:

```suggestion
                if float_compare(
                    order.invoice_id.amount_total_signed,
                    order.amount_total,
                    precision_digits=prec
                ):
```

It compares arg1 vs arg2 with the provided precision
possible return values are: `-1` (it's lower), `0` (equal) or `1` (it's bigger)

Reply:
you're right again ! 
sorry for the noice. Fixed.